Source/Gauntlet/Assertions/EqualWithAccuracyAsserts.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  EqualWithAccuracyAsserts.swift
+//
+//  MIT License
+//
+//  Copyright (c) [2023] The Kroger Co. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in all
+//  copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//  SOFTWARE.
+
+import Foundation
+
+extension Assertion where Value: Numeric {
+
+    /// Asserts that the value is equal to the specified value +/- the given accuracy range.
+    ///
+    /// - Parameters:
+    ///   - expectedValue: The value the receiver should be equal to.
+    ///   - accuracy: The maximum difference between the values for them to be considered equal.
+    ///
+    /// - Returns: A `Void` ``Assertion`` describing the result.
+    @discardableResult
+    public func isEqualTo(_ expectedValue: Value, accuracy: Value, line: Int = #line) -> Assertion<Void> {
```

Comment:
I don't think we should ever return `Assertion<Void>` (even for `Bool` 😬 ) so not to block any possible downstream assertions. They should all either pass through or return some transformed value.

Reply:
I see where you're coming from on this, but I think the context of these functions are important: These are not just general purpose functions, these are meant for creating assertions in test code. When you have asserted that a value is `true` or `false` there is *nothing* left to do with that value. By returning an` Assertion<Void>` you can use a `.then` closure to perform additional assertions, but nothing else. The `then` operator is the only operator that returns nothing and stops the chain. 

If we returned an `Assertion<Bool>` and a `then` closure was used the user would then have to explicitly ignore, or define a value for, the incoming Bool value provided to the closure. But it was already validated so that value is *completely meaningless* in the context of assertions. There is no meaningful downstream assertion that can be blocked.

I think the same is true for equality. You have fully validated the value, there is nothing left to do at that point. If you're in a case where you have a type that conforms to equality with some subset of it's values, then doing an `Assert(that: value).isEqualTo(expectedEqualValue)` should be one of multiple assertions you're running, not the first in a chain of assertions.